docs/pages/admin-guides/deploy-a-cluster/separate-proxy-service-endpoints.mdx

```diff
@@ -0,0 +1,196 @@
+---
+title: Separate Internal and External Proxy Service Traffic
+description: Explains how to set up the Teleport Proxy Service to isolate traffic from the public internet from internal client traffic.
+---
+
+In a typical Teleport cluster, including clusters deployed on Teleport Cloud,
+the Teleport Proxy Service exposes a single HTTPS port to users on the public
+internet as well as infrastructure resources running in private networks. For
+self-hosted Teleport clusters with security needs that require it, you can
+configure your cluster to handle traffic to the Proxy Service differently
+depending on whether it originates from the public internet or from your
+internal networks.
+
+This guide explains how to set up a self-hosted Teleport cluster to separate
+Proxy Service HTTPS traffic for internal and external clients.
+
+(!docs/pages/includes/cloud/call-to-action.mdx!)
+
+## Architecture
+
+To handle Teleport Proxy Service traffic differently depending on the origin,
+you can deploy separate Proxy Service load balancers:
+
+- A load balancer for end-user traffic from the public internet
+- A load balancer for traffic from Teleport Agents in private networks
+
+Each load balancer has a separate IP address. Both load balancers route traffic
+to the same pool of Teleport Proxy Service instances, such as the same AWS
+target group.
+
+Clients then connect to the Proxy Service load balancer at the appropriate
+address. Teleport Agents establish SSH reverse tunnels to the Proxy Service by
+dialing the internal load balancer, while external clients connect to the Proxy
+Service using the external load balancer. 
+
+When an end user connects to an infrastructure resource, e.g., using `tsh proxy
+db` or sending an HTTP request to a Teleport-protected application, they dial
+the external load balancer, which forwards the traffic to the Teleport Proxy
+Service. The Proxy Service then routes the traffic to the appropriate
+infrastructure resource by selecting a reverse tunnel.
+
+## Routing traffic to the appropriate load balancer
+
+Assuming you have deployed two load balancers, one for external Proxy Service
+clients and one for internal clients, there are three ways strategies for
+routing traffic to the appropriate load balancer from Teleport Agents and end
+users:
+
+- [Split DNS](#split-dns)
+- [Multiple public addresses](#multiple-public-addresses)
+- [Tunnel public address](#tunnel-public-address)
+
+### Split DNS
+
+You can set up your infrastructure to use split DNS so that a single public
+address for your Teleport cluster resolves to the external endpoint for end
+users and the internal endpoint for infrastructure resources in your internal
+network.
```

Comment:
Can we say that this is the _strongly recommended_ option. Option #2 requires setting an env var on every agent which is cumbersome, and error-prone. Option #3 has the users disable TLS multiplexing which is the opposite of what we want, we are currently strongly pushing for every deployment to use TLS multiplexing as this heavily reduces the risks of something going wrong in the network setup, is easier to troubleshoot, and offers better/faster/more reliable connectivity.

Reply:
Agree with @hugoShaka here.  Split DNS to me is the easiest and cleanest approach and should always be the first option presented.  The others are only there because the concept of split DNS is polarizing with customers at times.